ultraplot/tests/test_colorbar.py

```diff
@@ -353,3 +354,135 @@ def test_label_placement_fig_colorbar2():
     fig, axs = uplt.subplots(nrows=1, ncols=2)
     fig.colorbar(cmap, loc="bottom", label="My Label", labelloc="right")
     return fig
+
+
+@pytest.mark.parametrize(
+    ("labelloc", "cbarloc"),
+    product(
+        ["bottom", "top", "left", "right"],
+        [
+            "top",
+            "bottom",
+            "left",
+            "right",
+            "upper right",
+            "upper left",
+            "lower left",
+            "lower right",
+        ],
+    ),
+)
+def test_colorbar_label_placement(labelloc, cbarloc):
+    """
+    Ensure that colorbar label placement works correctly.
+    """
+    cmap = uplt.Colormap("plasma_r")
+    title = "My Label"
+    fig, ax = uplt.subplots()
+
+    cbar = ax.colorbar(cmap, loc=cbarloc, labelloc=labelloc, title=title)
+
+    x_label = cbar.ax.xaxis.label.get_text()
+    y_label = cbar.ax.yaxis.label.get_text()
+
+    assert title in (x_label, y_label), (
+        f"Expected label '{title}' not found. "
+        f"xaxis label: '{x_label}', yaxis label: '{y_label}', "
+        f"labelloc='{labelloc}', cbarloc='{cbarloc}'"
+    )
```

Comment:
This assert needs to be specialized to the specific test case, only testing one of x or y.

Reply:
You are correct that the locations may differ, but in the end we are only attaching it to x or y depending on the location. Checking for if the label exists would imply that the label is set correctly otherwise an error would be raised.